frontend/packages/knative-plugin/src/components/add/channels/ChannelForm.tsx

```diff
@@ -0,0 +1,96 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { FormikProps, FormikValues, useFormikContext } from 'formik';
+import { FormFooter, FlexForm, useFormikValidationFix } from '@console/shared';
+import { LoadingInline } from '@console/internal/components/utils';
+import {
+  isDefaultChannel,
+  getChannelKind,
+  getChannelData,
+  useDefaultChannelConfiguration,
+} from '../../../utils/create-channel-utils';
+import { ChannelListProps } from '../import-types';
+import YamlSection from './sections/YamlSection';
+import FormViewSection from './sections/FormViewSection';
+import ChannelSelector from './form-fields/ChannelSelector';
+
+interface OwnProps {
+  namespace: string;
+  channels: ChannelListProps;
+}
+
+const ChannelForm: React.FC<FormikProps<FormikValues> & OwnProps> = ({
+  errors,
+  handleSubmit,
+  handleReset,
+  status,
+  isSubmitting,
+  dirty,
+  namespace,
+  channels,
+}) => {
+  const {
+    values,
+    setFieldValue,
+    setFieldTouched,
+    validateForm,
+    setErrors,
+    setStatus,
+  } = useFormikContext<FormikValues>();
+  useFormikValidationFix(values);
+  const [defaultChannel, defaultChannelLoaded] = useDefaultChannelConfiguration(namespace);
+  const channelHasFormView = values.type && isDefaultChannel(getChannelKind(values.type));
+  const channelKind = getChannelKind(values.type);
+  const onTypeChange = React.useCallback(
+    (item: string) => {
+      setErrors({});
+      setStatus({});
+      const kind = getChannelKind(item);
+      if (isDefaultChannel(kind) && kind.length > 0) {
```

Comment:
nit: do we check for `kind.length > 0` as well as `isDefaultChannel(kind)` will return `true` if kind is there of default type

Reply:
removed it.